### PR TITLE
Fix outdated parameters for :class:`LinearTransformationScene` and add an example + typing.

### PR DESCRIPTION
**Pull request by hydrobeam**
## Motivation

The defaults for the foreground number plane in `LinearTransformationScene` were using the previous `x_min, x_max` method of setting the size of the plane. Moreover, it was not possible to adjust any properties for the foreground plane. This PR fixes those issues and adds typing + a basic example.

## Explanation for Changes

[Documentation Reference](https://manimce--1545.org.readthedocs.build/en/1545/reference/manim.scene.vector_space_scene.LinearTransformationScene.html)

Brought in the static method `update_default_config`, from `coordinate_systems.py` so that values can be adjusted by the user for `foreground/background plane_kwargs`.

Also removed `self.t_matrix` from the default properties of the scene as this matrix was not used anywhere.

## Further Comments

There's a decent amount of messiness in this Scene that should probably be patched up eventually.

## Checklist
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)
- [x] I have written a descriptive PR title (see top of PR template for examples)
- [x] I have written a changelog entry for the PR or deem it unnecessary
- [x] My new functions/classes either have a docstring or are private
- [x] My new functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [x] My new documentation builds, looks correctly formatted, and adds no additional build warnings

## Reviewer Checklist
- [ ] The PR title is descriptive enough
- [ ] The PR is labeled correctly
- [ ] The changelog entry is completed if necessary
- [ ] Newly added functions/classes either have a docstring or are private
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings

### CODE REVIEW

**Review by Nikhil0504 — APPROVED** [review]
LGTM and update the branch up to date

File: manim/scene/vector_space_scene.py [hunk 1]
```diff
@@ -499,53 +504,103 @@ def show_ghost_movement(self, vector):
 class LinearTransformationScene(VectorScene):
     """
     This scene contains special methods that make it
-    especially suitable for showing Linear Transformations.
+    especially suitable for showing linear Transformations.
```
Comment (behackl) [inline manim/scene/vector_space_scene.py:507]:
```suggestion
    especially suitable for showing linear transformations.
```

File: manim/scene/vector_space_scene.py [hunk 2]
```diff
@@ -499,53 +504,103 @@ def show_ghost_movement(self, vector):
...
+    include_foreground_plane
+        Whether or not to include the foreground plane in the scene.
+    background_plane_kwargs
+        Parameters to be passed to :class:`NumberPlane` to adjust the background plane.
+    foreground_plane_kwargs
+        Parameters to be passed to :class:`NumberPlane` to adjust the foreground plane.
+    show_coordinates
+        Whether or not to include the coordinates for the background plane.
+    show_basis_vectors
+        Whether to show the basis x_axis -> `i_hat`` and y_axis -> `j_hat` vectors.
```
Comment (behackl) [inline manim/scene/vector_space_scene.py:522]:
```suggestion
        Whether to show the basis x_axis -> ``i_hat`` and y_axis -> ``j_hat`` vectors.
```

File: manim/scene/vector_space_scene.py [hunk 3]
```diff
@@ -499,53 +504,103 @@ def show_ghost_movement(self, vector):
...
+    basis_vector_stroke_width
+        The ``stroke_width`` of the basis vectors.
+    i_hat_color
+        The color of the `i_hat` vector.
+    j_hat_color
+        The color of the `j_hat` vector.
```
Comment (behackl) [inline manim/scene/vector_space_scene.py:528]:
```suggestion
        The color of the ``i_hat`` vector.
    j_hat_color
        The color of the ``j_hat`` vector.
```

File: manim/scene/vector_space_scene.py [hunk 4]
```diff
@@ -558,7 +613,6 @@ def setup(self):
         self.transformable_labels = []
         self.moving_mobjects = []
 
-        self.t_matrix = np.array(self.t_matrix)
```
Comment (behackl) [inline manim/scene/vector_space_scene.py:561]: Is this not used anywhere?

**Review by behackl — CHANGES_REQUESTED** [review]
Looks good to me overall, just some minor comments.

File: manim/scene/vector_space_scene.py (hunk 4 above)
Comment (hydrobeam) [reply manim/scene/vector_space_scene.py:561]: Nope, it was defined arbitrarily. Not relied on anywhere else.

**Review by behackl — APPROVED** [review]

**Pull request merged**